native/membrane_videocompositor_wgpu/src/compositor.rs

```diff
@@ -0,0 +1,262 @@
+            label: Some("sampler bind group"),
+            layout: &sampler_bind_group_layout,
+            entries: &[wgpu::BindGroupEntry {
+                binding: 0,
+                resource: wgpu::BindingResource::Sampler(&sampler),
+            }],
+        });
+
+        let shader_module = device.create_shader_module(wgpu::include_wgsl!("shader.wgsl"));
+
+        let pipeline_layout = device.create_pipeline_layout(&wgpu::PipelineLayoutDescriptor {
+            label: Some("pipeline"),
+            bind_group_layouts: &[&texture_bind_group_layout, &sampler_bind_group_layout],
+            push_constant_ranges: &[],
+        });
+
+        let pipeline = device.create_render_pipeline(&wgpu::RenderPipelineDescriptor {
+            label: Some("pipeline"),
+            layout: Some(&pipeline_layout),
+            primitive: wgpu::PrimitiveState {
+                topology: wgpu::PrimitiveTopology::TriangleList,
+                front_face: wgpu::FrontFace::Ccw,
+                cull_mode: Some(wgpu::Face::Back),
+                strip_index_format: None,
+                polygon_mode: wgpu::PolygonMode::Fill,
+                unclipped_depth: false,
+                conservative: false,
+            },
+            vertex: wgpu::VertexState {
+                module: &shader_module,
+                entry_point: "vs_main",
+                buffers: &[Vertex::LAYOUT],
+            },
+            fragment: Some(wgpu::FragmentState {
+                module: &shader_module,
+                entry_point: "fs_main",
+                targets: &[Some(wgpu::ColorTargetState {
+                    blend: None,
+                    write_mask: wgpu::ColorWrites::RED,
+                    format: wgpu::TextureFormat::R8Unorm,
+                })],
+            }),
+            multisample: wgpu::MultisampleState {
+                count: 1,
+                mask: !0,
+                alpha_to_coverage_enabled: false,
+            },
+            multiview: None,
+            depth_stencil: None,
+        });
+
+        Self {
+            device,
+            input_videos,
+            output_textures,
+            pipeline,
+            queue,
+            _sampler: sampler,
+            sampler_bind_group,
+        }
+    }
+
+    pub async fn join_frames(
+        &self,
+        upper_frame: &[u8],
+        lower_frame: &[u8],
+        output_buffer: &mut [u8],
+    ) {
+        self.input_videos[0].upload_data(&self.queue, upper_frame);
+        self.input_videos[1].upload_data(&self.queue, lower_frame);
+
+        let mut encoder = self
+            .device
+            .create_command_encoder(&wgpu::CommandEncoderDescriptor {
+                label: Some("encoder"),
+            });
+
+        for plane in [YUVPlane::Y, YUVPlane::U, YUVPlane::V] {
+            let mut render_pass = encoder.begin_render_pass(&wgpu::RenderPassDescriptor {
+                label: Some("i dont know yet"),
```

Comment:
Change `label: Some("i dont know yet")` to something like `label: Some("render pass")`